scalalib/src/mill/scalalib/cosmopolitan/CosmopolitanModule.scala

```diff
@@ -0,0 +1,135 @@
+package mill.scalalib.cosmopolitan
+
+import mill.api.{PathRef, Result}
+import mill.define.{Target => T, _}
+import mill.scalalib._
+
+trait CosmopolitanModule extends mill.Module with AssemblyModule {
+  def forkArgs(argv0: String): Task[Seq[String]] = Task.Anon { Seq[String]() }
+
+  def cosmoccVersion: T[String] = Task { "" }
+
+  def cosmocc: T[PathRef] = Task(persistent = true) {
+    val version = if (cosmoccVersion().isEmpty) "" else s"-${cosmoccVersion()}"
+    val versionedCosmocc = s"cosmocc${version}"
+    val zip = Task.dest / s"${versionedCosmocc}.zip"
+    val dest = Task.dest / versionedCosmocc / "bin" / "cosmocc"
+
+    if (!os.exists(dest)) {
+      os.write.over(
+        zip,
+        requests.get.stream(s"https://cosmo.zip/pub/cosmocc/${versionedCosmocc}.zip")
+      )
+      os.remove.all(Task.dest / versionedCosmocc)
+      os.call(("unzip", zip, "-d", Task.dest / versionedCosmocc))
```

Comment:
Let's use `os.unzip` instead.

Reply:
I did try `os.unzip` but it doesn't preserve permissions. Would have to have an extra step to give `cosmocc` executable permission. `cosmoc.zip` also contains other files. I'm not sure if not having the original permissions changes anything (though unlikely).

Reply:
Let's go with `os.zip` then and `os.perms.set` to fix anything necessary, and open an issue on OS-Lib about `os.unzip` not preserving permissions so we can fix that

Reply:
ok, using a subprocess is probably fine for now, but can  you open an issie upstream